spec/rspec/core/reporter_spec.rb

```diff
@@ -172,6 +172,34 @@ module RSpec::Core
       end
     end
 
+    describe "#publish" do
+      let(:listener) { double("listener", :custom => nil) }
+      before do
+        reporter.register_listener listener, :custom, :start
+      end
+
+      it 'will send custom events to registered listeners' do
+        expect(listener).to receive(:custom).with(RSpec::Core::Notifications::NullNotification)
+        reporter.publish :custom
+      end
+
+      it 'will raise when encountering RSpec standard events' do
+        expect { reporter.publish :start }.to raise_error
```

Comment:
`raise_error` with no args is prone to false positives -- for example, if you typo'd `reporter` or `publish` it would never execute the code you are intending to test but the spec would pass.  I think we should always pass some arg to `raise_error` that specifies some aspect of the error that couldn't result from ruby raising a `NameError` or `ArgumentError`.